hypothesis-python/src/hypothesis/extra/redis.py

```diff
@@ -57,6 +57,16 @@ def __repr__(self) -> str:
             f"RedisExampleDatabase({self.redis!r}, expire_after={self._expire_after!r})"
         )
 
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, RedisExampleDatabase)
+            # I'm not sure redis.Redis implemens eq at all, so this is probably
+            # always false in practice.
+            and self.redis == other.redis
```

Comment:
I mean `object.__eq__` ~ `return self is other` is a valid implementation... and [it doesn't look like](https://github.com/redis/redis-py/blob/36619a507addc313c0245bf318e4678938fc5d44/redis/client.py#L114) they do anything fancier; maybe we should open an upstream issue?

Reply:
I keep forgetting that `is` implies `==`. I've just removed the comment, I'm not confident in asking upstream to add a sane equality definition here